packages/mobile/scripts/run_e2e.sh

```diff
@@ -1,78 +1,105 @@
-react-native-kill-packager || echo 'Failed to kill for some reason'
-echo "Start metro server"
-yarn react-native start &
-
-
-echo "Waiting for device to connect to Wifi, this is a good proxy the device is ready"
-until adb shell dumpsys wifi | grep "mNetworkInfo" |grep "state: CONNECTED"
-do
-  sleep 10
-done
-
-cp ../../node_modules/.bin/jest node_modules/.bin/
-
-yarn test:detox
-STATUS=$?
-
- # Retry on fail logic
-if [ $STATUS -ne 0 ]; then
-   echo "It failed once, let's try again"
-   yarn test:detox
-   STATUS=$?
-fi
+yarn react-native-kill-packager || echo "Failed to kill package manager, proceeding anyway"
+
+# Build the app and run it
+if [ $PLATFORM = "android" ]; then
+  echo "Using platform android"
+
+  if [ -z $ANDROID_SDK_ROOT ]; then
+    echo "No Android SDK root set"
+    exit 1
+  fi
+
+  if [[ ! $($ANDROID_SDK_ROOT/emulator/emulator -list-avds | grep ^$VD_NAME$) ]]; then
+    echo "AVD $VD_NAME not installed. Please install it or change the detox configuration in package.json"
+    echo "You can see the list of available installed devices with $ANDROID_SDK_ROOT/emulator/emulator -list-avds"
+    exit 1
+  fi
+
+  if [ "$RELEASE" = false ]; then
+    CONFIG_NAME="android.emu.debug"
+  else
+    CONFIG_NAME="android.emu.release"
+  fi
+
+  if [ "$FAST" = false ]; then
+    echo "Configuring the app"
+    ./scripts/run_app.sh -p $PLATFORM -b
+  fi
+
+  echo "Building detox"
+  yarn detox build -c $CONFIG_NAME
+
+  echo "Starting the metro server"
+  yarn react-native start &
+
+  NUM_DEVICES=`adb devices -l | wc -l`
+  if [ $NUM_DEVICES -gt 2 ]; then 
+    echo "Emulator already running or device attached. Please shutdown / remove first"
+    exit 1
+  fi
+
+  echo "Starting the emulator"
+  $ANDROID_SDK_ROOT/emulator/emulator -avd $VD_NAME -no-boot-anim &
+
+  echo "Waiting for device to connect to Wifi, this is a good proxy the device is ready"
+  until [ `adb shell dumpsys wifi | grep "mNetworkInfo" | grep "state: CONNECTED" | wc -l` -gt 0 ]
+  do
+    sleep 3 
+  done
+
+  CELO_TEST_CONFIG=e2e yarn detox test -c $CONFIG_NAME -a e2e/tmp/ --take-screenshots=failing --record-logs=failing --detectOpenHandles -l verbose
+  STATUS=$?
+
+elif [ $PLATFORM = "ios" ]; then
+  echo "Using platform ios"
+  echo "IOS e2e tests not currently supported"
+  exit 1
 
-if [ $STATUS -ne 0 ]; then
-   # TODO: upload e2e_run.log and attach the link
-   echo "Test failed"
 else
-   echo "Test passed"
+  echo "Invalid value for platform, must be 'android' or 'ios'"
+  exit 1
 fi
 
-react-native-kill-packager
-
-echo "Closing emulator"
-kill -s 9 `ps -a | grep "Nexus_5X_API_28_x86" | grep -v "grep"  | awk '{print $1}'`
+echo "Done test, cleaning up"
+yarn react-native-kill-packager
 
-echo "Closing pidcat"
-kill -s 9 `ps -a | grep "pidcat" | grep -v "grep"  | awk '{print $1}'`
+echo "Closing emulator (if active)"
+adb devices | grep emulator | cut -f1 | while read line; do adb -s $line emu kill; done
```

Comment:
Should we move this to the android only block?

Reply:
Good idea!